engine/access/rpc/backend/backend_scripts.go

```diff
@@ -205,3 +216,49 @@ func (b *backendScripts) tryExecuteScript(ctx context.Context, executorAddress s
 	}
 	return execResp.GetValue(), nil
 }
+
+func (b *backendScripts) tryExecuteScriptOnArchiveNode(ctx context.Context, executorAddress string, blockID flow.Identifier,
+	script []byte,
+	arguments [][]byte) ([]byte, error) {
+	req := &access.ExecuteScriptAtBlockIDRequest{
+		BlockId:   blockID[:],
+		Script:    script,
+		Arguments: arguments,
+	}
+	port, err := findPortFromAddress(executorAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	archiveClient, closer, err := b.connFactory.GetAccessAPIClientWithPort(executorAddress, port)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create client for archive node %s: %v", executorAddress, err)
+	}
+	defer closer.Close()
+	resp, err := archiveClient.ExecuteScriptAtBlockID(ctx, req)
+	if err != nil {
+		if status.Code(err) == codes.Unavailable {
```

Comment:
Probably not a part of this PR, but this codo should be in an interceptor and not here

Reply:
will do! https://github.com/onflow/flow-go/issues/4526